### PR TITLE
Improve notification mechanism

### PR DESCRIPTION
**Pull request by jakalkhalili**

### CODE REVIEW

File: pkg/controller/jenkins/notifications/slack/slack.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+type Field struct {
+	Title string `json:"title"`
+	Value string `json:"value"`
+	Short bool   `json:"short"`
+}
+
+func (s Slack) getStatusColor(logLevel v1alpha2.NotificationLevel) event.StatusColor {
+	switch logLevel {
+	case v1alpha2.NotificationLevelInfo:
+		return "#439FE0"
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/slack/slack.go]: as previously, please name those colors as constants

File: pkg/controller/jenkins/notifications/slack/slack.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	case v1alpha2.NotificationLevelWarning:
+		return "danger"
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/slack/slack.go]: what is the color of danger? ;)

File: pkg/controller/jenkins/notifications/smtp/smtp.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+		return errors.Errorf("SMTP password is empty in secret '%s/%s[%s]", e.Jenkins.Namespace, passwordSelector.Name, passwordSelector.Key)
+	}
+
+	mailer := gomail.NewDialer(s.config.SMTP.Server, s.config.SMTP.Port, usernameSecretValue, passwordSecretValue)
+	mailer.TLSConfig = &tls.Config{InsecureSkipVerify: s.config.SMTP.TLSInsecureSkipVerify}
+
+	var statusMessage string
+
+	reasons := strings.TrimRight(strings.Join(e.Reason.Short(), "</li><li>"), "<li>")
+	statusMessage = "<ul><li>"
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/smtp/smtp.go]: as in previous comments, this "rendering" require a separate function with unit tests

File: pkg/controller/jenkins/notifications/slack/slack.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	default:
+		return "#c8c8c8"
+	}
+}
+
+// Send is function for sending directly to API
+func (s Slack) Send(e event.Event) error {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/slack/slack.go]: this function would benefit for separation of concerns:
- the sending operation itself
- the preparation of a message before sending, that is testable with unit tests

File: pkg/controller/jenkins/jenkins_controller.go [hunk 5]
```diff
@@ -254,12 +259,11 @@ func (r *ReconcileJenkins) reconcile(request reconcile.Request, logger logr.Logg
...
-			Jenkins:         *jenkins,
-			Phase:           notifications.PhaseBase,
-			LogLevel:        v1alpha2.NotificationLogLevelInfo,
-			Message:         message,
-			MessagesVerbose: messages,
+		*r.notificationEvents <- event.Event{
+			Jenkins: *jenkins,
+			Phase:   event.PhaseBase,
+			Level:   v1alpha2.NotificationLevelInfo,
+			Reason:  event.NewPodRestartReason(event.OperatorSource, []string{message}, messages),
```
Comment (tomaszsek) [inline pkg/controller/jenkins/jenkins_controller.go]: reason.NewBaseConfigurationComplete(event.OperatorSource, message)

File: pkg/controller/jenkins/jenkins_controller.go [hunk 6]
```diff
@@ -304,12 +307,11 @@ func (r *ReconcileJenkins) reconcile(request reconcile.Request, logger logr.Logg
...
-			Jenkins:         *jenkins,
-			Phase:           notifications.PhaseUser,
-			LogLevel:        v1alpha2.NotificationLogLevelInfo,
-			Message:         message,
-			MessagesVerbose: messages,
+		*r.notificationEvents <- event.Event{
+			Jenkins: *jenkins,
+			Phase:   event.PhaseUser,
+			Level:   v1alpha2.NotificationLevelInfo,
+			Reason:  event.NewPodRestartReason(event.OperatorSource, []string{message}, messages),
```
Comment (tomaszsek) [inline pkg/controller/jenkins/jenkins_controller.go]: reason.NewUserConfigurationComplete(event.OperatorSource, message)

File: pkg/controller/jenkins/jenkins_controller.go (hunk 5 above)
Comment (tomaszsek) [reply pkg/controller/jenkins/jenkins_controller.go]: messages -> userMessages

File: pkg/controller/jenkins/jenkins_controller.go [hunk 7]
```diff
@@ -219,12 +225,11 @@ func (r *ReconcileJenkins) reconcile(request reconcile.Request, logger logr.Logg
...
-			Jenkins:         *jenkins,
-			Phase:           notifications.PhaseBase,
-			LogLevel:        v1alpha2.NotificationLogLevelWarning,
-			Message:         message,
-			MessagesVerbose: messages,
+		*r.notificationEvents <- event.Event{
+			Jenkins: *jenkins,
+			Phase:   event.PhaseBase,
+			Level:   v1alpha2.NotificationLevelWarning,
+			Reason:  event.NewPodRestartReason(event.OperatorSource, []string{message}, messages),
```
Comment (tomaszsek) [inline pkg/controller/jenkins/jenkins_controller.go]: messages -> baseMessages

File: pkg/controller/jenkins/notifications/event/event.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package event
+
+import "github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
+
+// Phase defines the type of configuration
```
[the remark this answers is not in the document]
Comment (tomaszsek) [reply pkg/controller/jenkins/notifications/event/event.go]: Phase defines the context where notification has been generated: base or user.

File: pkg/controller/jenkins/notifications/mailgun/mailgun.go [hunk 9]
```diff
@@ -39,53 +42,58 @@ const content = `
...
-		message := event.Message + "<ul>"
-		for _, msg := range event.MessagesVerbose {
-			message = message + "<li>" + msg + "</li>"
-		}
-		message = message + "</ul>"
-		statusMessage = message
-	} else {
-		statusMessage = event.Message
-	}
+	reasons := strings.TrimRight(strings.Join(event.Reason.Short(), "</li><li>"), "<li>")
```
[the remark this answers is not in the document]
Comment (tomaszsek) [reply pkg/controller/jenkins/notifications/mailgun/mailgun.go]:
```
if config.Verbose {
    reasons := strings.TrimRight(strings.Join(event.Reason.Verbose(), "</li><li>"), "<li>")
} else {
    reasons := strings.TrimRight(strings.Join(event.Reason.Short(), "</li><li>"), "<li>")
}
```

File: pkg/controller/jenkins/notifications/smtp/smtp.go (hunk 3 above)
Comment (jakalkhalili) [reply pkg/controller/jenkins/notifications/smtp/smtp.go]: tbd

File: pkg/controller/jenkins/notifications/slack/slack.go (hunk 4 above)
Comment (jakalkhalili) [reply pkg/controller/jenkins/notifications/slack/slack.go]: tbd

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 10]
```diff
@@ -402,11 +409,11 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureJenkinsMasterPod(meta metav1.O
...
-		*r.Notifications <- notifications.Event{
-			Jenkins:  *r.Configuration.Jenkins,
-			Phase:    notifications.PhaseBase,
-			LogLevel: v1alpha2.NotificationLogLevelInfo,
-			Message:  "Creating a new Jenkins Master Pod",
+		*r.Notifications <- event.Event{
+			Jenkins: *r.Configuration.Jenkins,
+			Phase:   event.PhaseBase,
+			Level:   v1alpha2.NotificationLevelInfo,
+			Reason:  reason.NewPodCreation(reason.KubernetesSource, []string{"Creating a new Jenkins Master Pod"}),
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: reason.OperatorSource

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 11]
```diff
@@ -457,7 +464,12 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureJenkinsMasterPod(meta metav1.O
 		for _, msg := range messages {
 			r.logger.Info(msg)
 		}
-		return reconcile.Result{Requeue: true}, r.Configuration.RestartJenkinsMasterPod()
+
+		restartReason := reason.NewPodRestart(
+			reason.OperatorSource,
+			messages,
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: verbose messages are missing

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 12]
```diff
@@ -723,8 +735,14 @@ func (r *ReconcileJenkinsBaseConfiguration) waitForJenkins(meta metav1.ObjectMet
 	containersReadyCount := 0
 	for _, containerStatus := range jenkinsMasterPod.Status.ContainerStatuses {
 		if containerStatus.State.Terminated != nil {
-			r.logger.Info(fmt.Sprintf("Container '%s' is terminated, status '%+v', recreating pod", containerStatus.Name, containerStatus))
-			return reconcile.Result{Requeue: true}, r.Configuration.RestartJenkinsMasterPod()
+			message := fmt.Sprintf("Container '%s' is terminated, status '%+v', recreating pod", containerStatus.Name, containerStatus)
+			r.logger.Info(message)
+
+			restartReason := reason.NewPodRestart(
+				reason.OperatorSource,
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: reason.KubernetesSource

File: pkg/controller/jenkins/notifications/reason/reason.go [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+// Verbose is list of reasons with details
+func (p Undefined) Verbose() []string {
+	return append([]string{fmt.Sprintf("pod restarted by: '%s'", p.source)}, p.verbose...)
+}
+
+// TODO: test this
+func copyToVerboseIfNil(short []string, verbose []string) {
+	if reflect.DeepEqual(verbose, []string{}) || verbose == nil {
```
Comment (tomaszsek) [inline pkg/controller/jenkins/notifications/reason/reason.go]: if verbose == nil || len(verbose) == 0

File: pkg/controller/jenkins/notifications/reason/reason.go [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+		verbose: verbose,
+	}}
+}
+
+// Source is enum type that informs us what triggered notification
+type Source string
+
+// Short is list of reasons
+func (p Undefined) Short() []string {
+	return append([]string{fmt.Sprintf("pod restarted by: '%s'", p.source)}, p.short...)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/notifications/reason/reason.go]: return short

File: pkg/controller/jenkins/notifications/reason/reason.go [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+// Verbose is list of reasons with details
+func (p Undefined) Verbose() []string {
+	return append([]string{fmt.Sprintf("pod restarted by: '%s'", p.source)}, p.verbose...)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/notifications/reason/reason.go]: return verbose

File: pkg/controller/jenkins/notifications/sender.go [hunk 16]
```diff
@@ -3,130 +3,83 @@ package notifications
...
+						logger.Error(nil, fmt.Sprintf("%s", wrapped))
 					}
 				}
 			}(notificationConfig)
 		}
-		k8sEvent.Emit(&evt.Jenkins, logLevelEventType(evt.LogLevel), "NotificationSent", evt.Message)
+		k8sEvent.Emit(&e.Jenkins,
+			eventLevelToKubernetesEventType(e.Level),
+			k8sevent.Reason(reflect.TypeOf(e.Reason).Name()),
+			strings.Join(e.Reason.Short(), ";"),
```
Comment (tomaszsek) [inline pkg/controller/jenkins/notifications/sender.go]: It will be too long for k8s events. `e.Reason.Short()[0]` will be enough.

File: pkg/controller/jenkins/notifications/mailgun/mailgun.go [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+		return infoColor
+	case v1alpha2.NotificationLevelWarning:
+		return warningColor
+	default:
+		return defaultColor
+	}
+}
+
+// Send is function for sending directly to API
+func (m MailGun) Send(event event.Event) error {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/mailgun/mailgun.go]: I would separate the concerns of sending from message creation to simplify code for readability and testing

File: pkg/controller/jenkins/notifications/msteams/msteams.go [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/provider"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	infoColor    = "439FE0"
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/msteams/msteams.go]: :+1:

File: pkg/controller/jenkins/notifications/msteams/msteams.go [hunk 19]
```diff
@@ -0,0 +1,143 @@
...
+		return infoColor
+	case v1alpha2.NotificationLevelWarning:
+		return warningColor
+	default:
+		return defaultColor
+	}
+}
+
+// Send is function for sending directly to API
+func (t Teams) Send(e event.Event) error {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/msteams/msteams.go]: I would separate the concerns of sending from message creation to simplify code for readability and testing

File: pkg/controller/jenkins/notifications/reason/reason.go [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+	return p.short
+}
+
+// Verbose is list of reasons with details
+func (p Undefined) Verbose() []string {
+	return p.verbose
+}
+
+func copyToVerboseIfNil(short []string, verbose *[]string) {
+	if verbose == nil || len(*verbose) == 0 {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/reason/reason.go]: I would check if short is nil and throw error or panic

File: pkg/controller/jenkins/notifications/reason/reason_test.go [hunk 21]
```diff
@@ -0,0 +1,19 @@
+package reason
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCopyToVerboseIfNil(t *testing.T) {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/reason/reason_test.go]: please test edge cases, not only happy path

File: pkg/controller/jenkins/notifications/slack/slack.go [hunk 22]
```diff
@@ -0,0 +1,142 @@
...
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/provider"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	infoColor    = "#439FE0"
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/slack/slack.go]: :+1:

File: pkg/controller/jenkins/notifications/smtp/smtp.go [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+	"gopkg.in/gomail.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	mailSubject = "Jenkins Operator Notification"
+
+	infoColor    = "blue"
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/smtp/smtp.go]: :+1:

File: pkg/controller/jenkins/notifications/smtp/smtp.go [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+	config    v1alpha2.Notification
+}
+
+// New returns instance of SMTP
+func New(k8sClient k8sclient.Client, config v1alpha2.Notification) *SMTP {
+	return &SMTP{k8sClient: k8sClient, config: config}
+}
+
+// Send is function for sending notification by SMTP server
+func (s SMTP) Send(e event.Event) error {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/smtp/smtp.go]: I would separate the concerns of sending from message creation to simplify code for readability and testing

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 25]
```diff
@@ -548,10 +565,11 @@ func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMa
 			messages = append(messages, fmt.Sprintf("Container '%+v' not found in pod, recreating pod", actualContainer))
 		}
 
-		messages = append(messages, r.compareContainers(*expectedContainer, actualContainer)...)
+		messages = append(messages, "Jenkins master pod container attributes has changed")
+		verbose = append(verbose, r.compareContainers(*expectedContainer, actualContainer)...)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: `r.compareContainers` should return messages and verbose messages

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 26]
```diff
@@ -502,37 +512,44 @@ func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMa
 	}
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: should return with `reason.KubernetesSource`

File: pkg/controller/jenkins/configuration/base/reconcile.go (hunk 12 above)
Comment (tomaszsek) [reply pkg/controller/jenkins/configuration/base/reconcile.go]: ?

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 27]
```diff
@@ -480,8 +488,9 @@ func isPodTerminating(pod corev1.Pod) bool {
 	return pod.ObjectMeta.DeletionTimestamp != nil
 }
 
-func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMasterPod corev1.Pod, userAndPasswordHash string) []string {
+func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMasterPod corev1.Pod, userAndPasswordHash string) *reason.PodRestart {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: I would use the interface as the return type

Comment (pawelprazak) [reply pkg/controller/jenkins/configuration/base/reconcile.go]: also the name starting with `is...` would suggest that the return type would be `bool` which is no longer the case,
 my suggestion is `checkForPodRecreation`

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 28]
```diff
@@ -452,12 +459,13 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureJenkinsMasterPod(meta metav1.O
 		return reconcile.Result{Requeue: true}, nil
 	}
 
-	messages := r.isRecreatePodNeeded(*currentJenkinsMasterPod, userAndPasswordHash)
-	if hasMessages := len(messages) > 0; hasMessages {
-		for _, msg := range messages {
+	restartReason := r.isRecreatePodNeeded(*currentJenkinsMasterPod, userAndPasswordHash)
+	if hasMessages := len(restartReason.Short()) > 0; hasMessages {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: I would replace it with a simple `nil` check

File: pkg/controller/jenkins/notifications/mailgun/mailgun.go [hunk 29]
```diff
@@ -0,0 +1,116 @@
...
+
+	secretValue := string(secret.Data[selector.Key])
+	if secretValue == "" {
+		return errors.Errorf("Mailgun API secret is empty in secret '%s/%s[%s]",
+			event.Jenkins.Namespace, selector.Name, selector.Key)
+	}
+
+	mg := mailgun.NewMailgun(m.config.Mailgun.Domain, secretValue)
+
+	msg := mg.NewMessage(fmt.Sprintf("Jenkins Operator Notifier <%s>",
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/mailgun/mailgun.go]: this line is not clear to me, maybe naming arguments as variables would help

File: pkg/controller/jenkins/notifications/mailgun/mailgun.go [hunk 30]
```diff
@@ -0,0 +1,116 @@
...
+	case v1alpha2.NotificationLevelInfo:
+		return infoColor
+	case v1alpha2.NotificationLevelWarning:
+		return warningColor
+	default:
+		return defaultColor
+	}
+}
+
+func (m MailGun) generateMessage(event event.Event) string {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/mailgun/mailgun.go]: are there any unit tests for this?

File: pkg/controller/jenkins/notifications/reason/reason_test.go [hunk 31]
```diff
@@ -0,0 +1,32 @@
...
+		var verbose []string
+		short := []string{"test", "string"}
+
+		copyToVerboseIfNil(short, &verbose)
+
+		assert.NotNil(t, verbose)
+		assert.Equal(t, short, verbose)
+	})
+
+	t.Run("copy valid first, then invalid", func(t *testing.T) {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/reason/reason_test.go]: how about some empty values or `nil`'s

File: pkg/controller/jenkins/notifications/slack/slack.go [hunk 32]
```diff
@@ -0,0 +1,152 @@
...
+	case v1alpha2.NotificationLevelInfo:
+		return infoColor
+	case v1alpha2.NotificationLevelWarning:
+		return warningColor
+	default:
+		return defaultColor
+	}
+}
+
+func (s Slack) generateMessage(e event.Event) Message {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/slack/slack.go]: unit tests?

File: pkg/controller/jenkins/notifications/smtp/smtp.go [hunk 33]
```diff
@@ -0,0 +1,134 @@
...
+	k8sClient k8sclient.Client
+	config    v1alpha2.Notification
+}
+
+// New returns instance of SMTP
+func New(k8sClient k8sclient.Client, config v1alpha2.Notification) *SMTP {
+	return &SMTP{k8sClient: k8sClient, config: config}
+}
+
+func (s SMTP) generateMessage(e event.Event) *gomail.Message {
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/smtp/smtp.go]: unit tests?

File: pkg/controller/jenkins/notifications/mailgun/mailgun_test.go [hunk 34]
```diff
@@ -0,0 +1,174 @@
...
+		}
+
+		var statusMessage strings.Builder
+		r := strings.TrimRight(strings.Join(e.Reason.Verbose(), "</li><li>"), "<li>")
+
+		statusMessage.WriteString("<ul><li>")
+		statusMessage.WriteString(r)
+		statusMessage.WriteString("</ul>")
+
+		message := s.generateMessage(e)
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/mailgun/mailgun_test.go]: message -> got

File: pkg/controller/jenkins/notifications/mailgun/mailgun_test.go [hunk 35]
```diff
@@ -0,0 +1,174 @@
...
+
+		assert.Equal(t, message, fmt.Sprintf(content, s.getStatusColor(e.Level),
```
Comment (pawelprazak) [inline pkg/controller/jenkins/notifications/mailgun/mailgun_test.go]: fmt.Sprintf(...) -> got

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 36]
```diff
@@ -480,59 +488,74 @@ func isPodTerminating(pod corev1.Pod) bool {
...
+		currentJenkinsMasterPod.Status.Phase == corev1.PodSucceeded ||
+		currentJenkinsMasterPod.Status.Phase == corev1.PodUnknown {
+		messages = append(messages, fmt.Sprintf("Invalid Jenkins pod phase '%s'", currentJenkinsMasterPod.Status.Phase))
+		verbose = append(verbose, fmt.Sprintf("Invalid Jenkins pod phase '%+v'", currentJenkinsMasterPod.Status))
+		return reason.NewPodRestart(reason.KubernetesSource, messages, verbose...)
+	}
+
 	if userAndPasswordHash != r.Configuration.Jenkins.Status.UserAndPasswordHash {
-		messages = append(messages, "User or password have changed, recreating pod")
+		messages = append(messages, "User or password have changed")
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: verbose message is missing

File: pkg/controller/jenkins/configuration/base/reconcile.go [hunk 37]
```diff
@@ -480,59 +488,74 @@ func isPodTerminating(pod corev1.Pod) bool {
...
 	}
 
 	if r.Configuration.Jenkins.Spec.Restore.RecoveryOnce != 0 && r.Configuration.Jenkins.Status.RestoredBackup != 0 {
-		messages = append(messages, "spec.restore.recoveryOnce is set, recreating pod")
+		messages = append(messages, "spec.restore.recoveryOnce is set")
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/reconcile.go]: verbose message is missing
